# Project Name : Customer Segment Prediction


## Name : Sushilkumar Dhepe

# Importing Libraries


In [2]:
import pandas as pd
from sklearn.preprocessing import LabelEncoder
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import  confusion_matrix,classification_report

# Data gathering from the Kaggle 

In [3]:
Data = pd.read_csv("ELD Service Provider Segmentation Dataset.csv")
# Explore Dataset
print(Data.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2333 entries, 0 to 2332
Data columns (total 6 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   Segment                2333 non-null   object 
 1   Territory              2333 non-null   object 
 2   Customer Date          2333 non-null   object 
 3   Net ARR                2332 non-null   float64
 4   Composite Fleet Size   2333 non-null   int64  
 5   Number of ELD Devices  2331 non-null   float64
dtypes: float64(2), int64(1), object(3)
memory usage: 109.5+ KB
None


# Data Preprocessing
1. Data Cleaning
2. Checking Duplicate Data
3. Encoding Data
4. Removing irrelavent Data/Columns

In [3]:
 # 1/ Cleaning the data from Missing values
print("\n Missing values : ")
print(Data.isna().sum())
Data.dropna(axis =0 , inplace=True)

# 2/ Check duplicates
print("\n Duplicates  :")
print(Data.duplicated().sum())

# 3/ Encoding Data
labelencoder=LabelEncoder()
Data['Territory'] = labelencoder.fit_transform(Data['Territory'])
Data['Segment'] = labelencoder.fit_transform(Data['Segment'])


# 4/ Removing Unwanted columns
print("\n",Data.corr(),"\n")
Data.drop(columns= "Customer Date", inplace=True)

print(Data.head())


 Missing values : 
Segment                  0
Territory                0
Customer Date            0
Net ARR                  1
Composite Fleet Size     0
Number of ELD Devices    2
dtype: int64

 Duplicates  :
0

                         Segment  Territory   Net ARR  Composite Fleet Size  \
Segment                1.000000  -0.024278  0.261434              0.143073   
Territory             -0.024278   1.000000  0.052178              0.056903   
Net ARR                0.261434   0.052178  1.000000              0.486156   
Composite Fleet Size   0.143073   0.056903  0.486156              1.000000   
Number of ELD Devices  0.274427   0.042400  0.935234              0.548331   

                       Number of ELD Devices  
Segment                             0.274427  
Territory                           0.042400  
Net ARR                             0.935234  
Composite Fleet Size                0.548331  
Number of ELD Devices               1.000000   

   Segment  Territory   Net ARR 

# Data Modeling

In [4]:
 # 1/ Splitting data
# X --> features , Y --> Label

X = Data.drop(columns='Segment')
Y = Data.Segment

  # ** Split Features & Label to train and test
X_train, X_test, y_train, y_test = train_test_split(X, Y, test_size = 0.20,  shuffle= True ,random_state= 1 )
 

In [5]:
 # 2/ Calling Logistic regression
log_reg = LogisticRegression()

  # 3/ Train data
log_reg.fit(X_train, y_train)

c:\Users\adi63\anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:763: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


LogisticRegression()

In [6]:
  # 4/ Predict
y_pred = log_reg.predict(X_test)

 # Confisuion Matrix
#print(confusion_matrix(y_test, y_pred))
# 5/ Report
report = classification_report(y_test, y_pred)
print('report:', report, sep='\n')


report:
              precision    recall  f1-score   support

           0       0.86      0.82      0.84       309
           1       0.44      0.47      0.45        15
           2       0.65      0.72      0.68       143

    accuracy                           0.78       467
   macro avg       0.65      0.67      0.66       467
weighted avg       0.79      0.78      0.78       467



### The results are nothing but accuracy of data that is through out the dataset 20% dataset is having 78% of accuracy.
